volatility3/framework/plugins/linux/lsblk.py

```diff
@@ -0,0 +1,447 @@
+            if inode.i_sb == blockdev_superblock and inode.i_rdev == devt:
+                break
+            hlist_node = hlist_node.next
+
+        return inode.i_bdev
+
+    def _device_iterator(self, vmlinux, klist_devices):
+
+        # Linux v5.1 moves device->knode_class to device_private->knode_class
+        knode_class_in_private_device = vmlinux.get_type("device_private").has_member(
+            "knode_class"
+        )
+
+        if knode_class_in_private_device:
+            knode_class_off = vmlinux.get_type("device_private").relative_child_offset(
+                "knode_class"
+            )
+            for klist_node in klist_devices.k_list.to_list(
+                vmlinux.symbol_table_name + constants.BANG + "klist_node", "n_node"
+            ):
+                device_private = vmlinux.object(
+                    object_type="device_private",
+                    offset=klist_node.vol.offset - knode_class_off,
+                    absolute=True,
+                )
+                yield device_private.device.dereference()
+        else:
+            knode_class_off = vmlinux.get_type("device").relative_child_offset(
+                "knode_class"
+            )
+            for klist_node in klist_devices.k_list.to_list(
+                vmlinux.symbol_table_name + constants.BANG + "klist_node", "n_node"
+            ):
+                device = vmlinux.object(
+                    object_type="device",
+                    offset=klist_node.vol.offset - knode_class_off,
+                    absolute=True,
+                )
+                yield device
+
+    def _get_name(self, vmlinux, block_device, gendisk):
+        # block_device does not have bd_partno before 5.11
+        if vmlinux.get_type("block_device").has_member("bd_partno"):
+            if block_device.bd_partno:
+                return utility.array_to_string(gendisk.disk_name) + str(
+                    block_device.bd_partno
+                )
+        else:
+            if gendisk.part0.partno:
+                return utility.array_to_string(gendisk.disk_name) + str(
+                    gendisk.part0.partno
+                )
+
+        if utility.array_to_string(gendisk.disk_name).startswith("dm-"):
+            if vmlinux.has_type("mapped_device"):
+                mapped_device = vmlinux.object(
+                    object_type="mapped_device",
+                    offset=gendisk.private_data,
+                    absolute=True,
+                )
+
+                hash_cell = vmlinux.object(
+                    object_type="hash_cell",
+                    offset=mapped_device.interface_ptr,
+                    absolute=True,
+                )
+
+                if hash_cell.name:
+                    return utility.pointer_to_string(hash_cell.name, 32)
+
+        return utility.array_to_string(gendisk.disk_name)
+
+    def _format_bytes(self, size):
```

Comment:
Feels like this could be used for more than just the `linux.lsblk` function, consider adding it as a staticmethod somewhere more common across plugins.  As mentioned above, this is really a UI function rather than a plugin function.  The plugin should return the raw data, and let the UI decide what needs to happen to display it properly (using a type hint to say how you'd expect it to be displayed).